packages/opentelemetry/lib/tracing/tracer.js

```diff
@@ -0,0 +1,89 @@
+/* eslint-disable no-unused-vars */
+
+import { TraceAPI as api, SpanStatusCode, SpanOptions } from '@opentelemetry/api';
+
+import { Span } from '@opentelemetry/tracing';
+
+class Tracer {
+
+  constructor (tracerProviderInstance, name, options = {}) {
+    this.tracingProvider = tracerProviderInstance;
+    const { version } = options;
+    this._tracer = api.trace.getTracer(name, version);
+  }
+
+  /**
+   * creates a span object with optional parent span and optional span options
+   * @param { string } name  name of the span
+   * @param { Span } parentSpan (Optional) parentSpan to get context from
+   * @param { SpanOptions } spanOptions (Optional) span options
+   * @return { Span }
+   */
+  createSpanObject (name, parentSpan = null, spanOptions = null) {
+    //TODO - verify setting context from parentSpan
+    const context = parentSpan ? api.setSpan(api.context.active(), parentSpan) : null;
+    return this._tracer.startSpan(name, spanOptions, context);
+  }
+
+
+  /**
+   * monkeypathces argument async function to instrument it via a span object
+   * @param { string } name  name of the span
```

Comment:
there should be no spaces between curly brackets. Read https://jsdoc.app/tags-param.html for more details